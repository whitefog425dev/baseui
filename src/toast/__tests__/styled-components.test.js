/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {StyledBody, StyledCloseIcon, KIND} from '../index';

describe('Component styled components', () => {
  test('StyledBody - basic render', () => {
    const component = shallow(<StyledBody />);

    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledBody has correct default styles',
    );

    component.setProps({$kind: KIND.info});
    expect(component.instance().getStyles().backgroundColor).toEqual(
      '$theme.colors.primary500',
    );

    component.setProps({$kind: KIND.positive});
    expect(component.instance().getStyles().backgroundColor).toEqual(
      '$theme.colors.positive500',
    );

    component.setProps({$kind: KIND.warning});
    expect(component.instance().getStyles().backgroundColor).toEqual(
      '$theme.colors.warning500',
    );

    component.setProps({$kind: KIND.negative});
    expect(component.instance().getStyles().backgroundColor).toEqual(
      '$theme.colors.negative500',
    );

    component.setProps({$isHidden: false, $isAnimating: false});
    expect(component.instance().getStyles().height).toEqual('auto');
    expect(component.instance().getStyles().opacity).toEqual(1);

    component.setProps({$isHidden: true, $isAnimating: true});
    expect(component.instance().getStyles().height).toEqual('auto');
    expect(component.instance().getStyles().opacity).toEqual(0);

    component.setProps({$isHidden: true, $isAnimating: false});
    expect(component.instance().getStyles().height).toEqual(0);
    expect(component.instance().getStyles().opacity).toEqual(0);
  });
  test('StyledCloseIcon - basic render', () => {
    const component = shallow(<StyledCloseIcon $prop={false} />);

    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledSvg has correct default styles',
    );
  });
});
