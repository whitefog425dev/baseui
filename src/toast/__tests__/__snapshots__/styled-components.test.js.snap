// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component styled components StyledBody - basic render: StyledBody has correct default styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.primary500",
  "borderRadius": "$theme.borders.radius200",
  "boxShadow": "$theme.lighting.shadow600",
  "color": "$theme.colors.toastText",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "height": "auto",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginTop": "$theme.sizing.scale300",
  "opacity": 0,
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
  "pointerEvents": "auto",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "all",
  "transitionTimingFunction": "$theme.animation.easeInOutCurve",
  "width": "288px",
}
`;

exports[`Component styled components StyledBody - inline type styles: StyledBody has correct default styles when type is set to TYPE.inline 1`] = `
Object {
  "backgroundColor": "$theme.colors.primary500",
  "borderRadius": "$theme.borders.radius200",
  "boxShadow": "$theme.lighting.shadow600",
  "color": "$theme.colors.toastText",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "height": "auto",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginTop": "$theme.sizing.scale300",
  "opacity": 0,
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
  "pointerEvents": "auto",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "all",
  "transitionTimingFunction": "$theme.animation.easeInOutCurve",
  "width": "288px",
}
`;

exports[`Component styled components StyledCloseIcon - basic render: StyledSvg has correct default styles 1`] = `
Object {
  "color": "currentColor",
  "cursor": "pointer",
  "display": "inline-block",
  "fill": "currentColor",
  "float": "right",
  "height": "$theme.sizing.scale600",
  "width": "$theme.sizing.scale600",
}
`;
